### PR TITLE
Add support for custom entrypoint

### PR DESCRIPTION
**Pull request by JasonEtco**
Hey @mheap 👋 this lil' PR adds support for a custom `entrypoint`, using an input. I made a couple of unrelated tweaks to use Toolkit variables instead of environment variables (`actions-toolkit` abstracts them) but I'm happy to revert those if you prefer!

Let me know what you think 💪

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,7 +22,7 @@ Toolkit.run(async (tools) => {
   tools.log.complete("NPM install run");
 
   const { code } = await require("@zeit/ncc")(
-    path.join(workspace, "index.js"),
+    path.join(tools.workspace, tools.inputs.entrypoint),
```
Comment (JasonEtco) [inline index.js]: GitHub Actions will fill in the default value of `entrypoint`, but if you prefer for explicit-ness, we can add:

```suggestion
    path.join(tools.workspace, tools.inputs.entrypoint || 'index.js'),
```

**Review by mheap — COMMENTED** [review]
LGTM! Let me run this tomorrow and I’ll merge and release if it goes smoothly

File: index.js (hunk 1 above)
Comment (mheap) [inline index.js]: @JasonEtco Would `entrypoint` in this context be the entrypoint for the running action (`auto-compile-node`) or the repo that the workflow is running in?

Comment (JasonEtco) [reply index.js]: It’d be the input from the workflow file! Check out the README change for an example. It’s just a string that’s passed in the workflow.

Comment (mheap) [reply index.js]: Ah, I think I've worked it out in a test run. Will add a comment below

**Comment by mheap** [comment]
Big 👍 for this PR in general. 

When I tried to use your branch on a test repo it's fails: https://github.com/mheap/github-action-fail-at-weekend/runs/562466561

Whilst running it on the current `master` branch it passes: https://github.com/mheap/github-action-fail-at-weekend/runs/562469498

I *think* it's because we default `entrypoint` to `index.js` which is (should be?) a reserved input for Actions. By changing the default in our `action.yml` the runner changes the entrypoint for the Docker container, and we get an error that `index.js` isn't an executable

With that in mind, should we use a different name for this input?

**Review by mheap — CHANGES_REQUESTED** [review]
Big 👍 for this PR in general. 

When I tried to use your branch on a test repo it's fails: https://github.com/mheap/github-action-fail-at-weekend/runs/562466561

Whilst running it on the current `master` branch it passes: https://github.com/mheap/github-action-fail-at-weekend/runs/562469498

I *think* it's because we default `entrypoint` to `index.js` which is (should be?) a reserved input for Actions. By changing the default in our `action.yml` the runner changes the entrypoint for the Docker container, and we get an error that `index.js` isn't an executable

With that in mind, should we use a different name for this input?

**Comment by JasonEtco** [comment]
Aha! You're totally right @mheap, especially given [this doc](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepswithentrypoint):

> **`jobs.<job_id>.steps.with.entrypoint`**
> 
> Overrides the Docker `ENTRYPOINT` in the Dockerfile, or sets it if one wasn't already specified. 

That is unexpected, and a little weird - but alas, we can just change the name like you suggested. How about:

* `action-entrypoint`
* `main`
* `main-file`
* `index`

**Comment by mheap** [comment]
`main` makes sense to me, given that's the name used in `action.yml` too

**Comment by JasonEtco** [comment]
> `main` makes sense to me, given that's the name used in `action.yml` too

Makes, sense good call - done!

**Pull request merged**